lib/runtime/errors/transaction_validity.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package errors
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// TransactionValidityError Information on a transaction's validity and, if valid,
+// on how it relates to other transactions. It is a result of the form:
+// Result<ValidTransaction, TransactionValidityError>
+type TransactionValidityError scale.VaryingDataType
+
+var (
+	errInvalidType     = errors.New("invalid validity type")
+	errInvalidResult   = errors.New("invalid error value")
+	errInvalidTypeCast = errors.New("invalid type cast")
+)
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (tve *TransactionValidityError) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*tve)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*tve = TransactionValidityError(vdt)
+	return nil
+}
+
+// Value will return the value from the underlying VaryingDataType
+func (tve *TransactionValidityError) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*tve)
+	return vdt.Value()
+}
+
+// Error will return the error underlying TransactionValidityError
+func (tve *TransactionValidityError) Error() string {
+	invalidTxn, ok := tve.Value().(InvalidTransaction)
+	if !ok {
+		unknownTxn, ok2 := tve.Value().(UnknownTransaction)
+		if !ok2 {
+			return errInvalidTypeCast.Error()
+		}
+		return unknownTxn.Error()
+	}
+	return invalidTxn.Error()
+}
+
+// NewTransactionValidityError is constructor for TransactionValidityError
+func NewTransactionValidityError() TransactionValidityError {
+	vdt, err := scale.NewVaryingDataType(NewInvalidTransaction(), NewUnknownTransaction())
+	if err != nil {
+		panic(err)
+	}
+	return TransactionValidityError(vdt)
+}
+
+var (
+	ErrInvalidTxn = errors.New("invalid transaction")
+	ErrUnknownTxn = errors.New("unknown transaction")
+)
+
+// UnmarshalTransactionValidity Takes the result of the validateTransaction runtime call and unmarshalls it
```

Comment:
supernit why the uppercase on the second word 😄 

```suggestion
// UnmarshalTransactionValidity takes the result of the validateTransaction runtime call and unmarshalls it
```